kernel/CMakeLists.txt

```diff
@@ -117,6 +117,35 @@ target_sources_ifdef(CONFIG_OBJ_CORE              kernel PRIVATE obj_core.c)
 
 if(${CONFIG_KERNEL_MEM_POOL})
   target_sources(kernel PRIVATE mempool.c)
+
+  # Import all custom HEAP_MEM_POOL minimum size requirements
+  import_kconfig(CONFIG_HEAP_MEM_POOL_MIN_SIZE_ ${DOTCONFIG} min_size_keys)
+
+  set(max_min_size 0)
+
+  foreach(min_size ${min_size_keys})
+    if(${${min_size}} GREATER ${max_min_size})
+      set(max_min_size ${${min_size}})
+    endif()
+  endforeach()
+
+  if(CONFIG_HEAP_MEM_POOL_SIZE LESS max_min_size)
+    # Only warn if default value 0 has been modified
+    if(NOT CONFIG_HEAP_MEM_POOL_SIZE EQUAL 0)
+      message(WARNING "
+      CONFIG_HEAP_MEM_POOL_SIZE is less than HEAP_MEM_POOL_MIN_SIZE:
```

Comment:
Good to warn the user, but I think would be even better if we also inform the user about which value we're going to use.

Reply:
> Good to warn the user, but I think would be even better if we also inform the user about which value we're going to use.

Sure, makes sense. I've now added this.